code/modules/mob/living/carbon/alien/special/facehugger.dm

```diff
@@ -1,281 +1,271 @@
+	//Check for helmet
+	if(ishuman(carbon_target))
+		var/mob/living/carbon/human/human_target = carbon_target
+		if(human_target.is_mouth_covered(head_only = TRUE))
+			human_target.visible_message("<span class='danger'>[src] smashes against [target]'s [human_target.head]!</span>", "<span class='userdanger'>[src] smashes against your [human_target.head]!</span>")
+			COOLDOWN_START(src, coupling_cooldown, couple_retry_time)
+			return FALSE
 
-#define MIN_ACTIVE_TIME 200 //time between being dropped and going idle
-#define MAX_ACTIVE_TIME 400
+	//Check for mask
+	if(carbon_target.wear_mask)
+		var/obj/item/clothing/mask = carbon_target.wear_mask
+		if(istype(mask, /obj/item/clothing/mask/facehugger_item))
+			return FALSE
+		if(carbon_target.dropItemToGround(mask))
+			carbon_target.visible_message("<span class='danger'>[src] tears [mask] off of [target]'s face!</span>", "<span class='userdanger'>[src] tears [mask] off of your face!</span>")
+		else
+			carbon_target.visible_message("<span class='danger'>[src] tries to tear [mask] off of [target]'s face, but fails!</span>", "<span class='userdanger'>[src] trys to tear [mask] off of your face, but fails!</span>")
+			COOLDOWN_START(src, coupling_cooldown, couple_retry_time)
+			return FALSE
 
-/obj/item/clothing/mask/facehugger
+	target.visible_message("<span class='danger'>[src] leaps at [target]'s face!</span>", "<span class='userdanger'>[src] leaps at your face!</span>")
+	var/obj/item/clothing/mask/facehugger_item/hugger_item = BecomeItem()
+	if(!carbon_target.wear_mask && carbon_target.equip_to_slot_if_possible(hugger_item, ITEM_SLOT_MASK, FALSE, TRUE, TRUE))
+		hugger_item.Attach(carbon_target)
+	return TRUE
+
+/**
+ * Turns the facehugger into an item, storing the mob inside of it.
+ *
+ * Proc to turn the facehugger into an item and store the mob inside said item.
+ * The item version will take the name, description, and appearance of the facehugger.
+ */
+/mob/living/simple_animal/hostile/facehugger/proc/BecomeItem()
+	if(!mask_facehugger)
+		mask_facehugger = new(src, src)
+	mask_facehugger.forceMove(loc)
+	forceMove(mask_facehugger)
+	//This needs to be here instead of being an initialization arg since these can change over time.
+	mask_facehugger.name = name
+	mask_facehugger.desc = desc
+	mask_facehugger.icon_state = icon_state
+	return mask_facehugger
+
+/**
+ * # Facehugger Item
+ *
+ * The storage item for facehuggers, used to allow them to be worn as masks.
+ *
+ * An item used to represent a facehugger while in the inventory of something.  It stores the mob inside of it,
+ * keeping it safe from death.  Upon being dropped, the item will release the mob and destroy itself.  Also worth
+ * noting, throwing this item will transfer the throw's momentum and target to the facehugger, allowing for seamless
+ * throwing.  It also handles impregnating the wearer and subsequently killing the facehugger as well.
+ */
+/obj/item/clothing/mask/facehugger_item
```

Comment:
couldnt this use the mob holder system?

Reply:
I originally looked into using the system when I initially made the new facehuggers, however, they have a lot of undesirable traits integrated into them.

Reply:
Could you elaborate on the issues with holder objects and all?